### PR TITLE
Fix grammar to make function return type optional

### PR DESCRIPTION
**Pull request by dneto0**
Part of #1219

### CODE REVIEW

File: wgsl/index.bs
Comment (alan-baker) [inline wgsl/index.bs:4022]: The spec style seems to prefer (or at least has a non-trivial number of instances) having:
```
function_header
  : FN IDENT PAREN_LEFT param_list PAREN_RIGHT (ARROW function_return_type_decl)?
```

Editorially we should decide which way to go, but I don't think it needs to hold up this PR.

**Review by alan-baker — APPROVED** [review]

**Review by kvark — APPROVED** [review]

**Pull request merged**
